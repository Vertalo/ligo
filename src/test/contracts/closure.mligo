(* Test whether closures retain values in CameLIGO *)

let test (k: int) : int =
  let j: int = k + 5 in
  let close: (int -> int) =
    fun (i: int) -> i + j
  in
  let j: int = 20 in (* Shadow original variable to see if value close'd *)
  close 20
