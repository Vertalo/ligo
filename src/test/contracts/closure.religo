/* Test whether closures retain values in ReasonLIGO */

let test = (k: int): int => {
  let j: int = k + 5;
  let close: (int => int) = (i: int) => i + j;

  let j: int = 20; /* Shadow original variable to see if value close'd */
  close(20);
};
