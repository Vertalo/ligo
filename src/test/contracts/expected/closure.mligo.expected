let test (k : int) : int =
  let j : int = k + 5 in
  let close : int -> int = fun (i : int) -> i + j in
  let j : int = 20 in
  close 20
